bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/JSONUtils.java

```diff
@@ -297,7 +297,8 @@ public static boolean isJSONObject(Object json) {
         }
 
         BType type = ((RefValue) json).getType();
-        return type.getTag() == TypeTags.JSON_TAG || type.getTag() == TypeTags.MAP_TAG;
+        int typeTag = type.getTag();
+        return typeTag == TypeTags.JSON_TAG || typeTag == TypeTags.MAP_TAG || typeTag == TypeTags.RECORD_TYPE_TAG;
```

Comment:
Not related to this PR, but returning true for `isJSONObject` if ` typeTag == TypeTags.JSON_TAG` is wrong right? And we need to check the map's constraint too assuming this is a generic method.

Reply:
Just checked this. It seems simple literals are in their original type, so it doesn't count as a JSON tag. For example, the following works as expected:
```ballerina
import ballerina/io;

public function main() {
    json j1 = 10;
    io:println(j1.name);
}
```
`j1` is a `Long` when you check at this point.

Reply:
Yeah but `typeTag == TypeTags.JSON_TAG` will never be true right?

Reply:
Ah just checked it.. yeah, you're right.. for JSON objects, it's `map<json>`.. I'll remove this condition and see..

Reply:
Removed it in 4b4eaeab6e39b5cf11c6e0dbd24855b4643cf582